### ISSUE TITLE
Manila may choke on dates with UTC offset

### ISSUE THREAD

**Issue by mandre** [gophercloud/gophercloud#2507]
While reviewing https://github.com/gophercloud/gophercloud/pull/2496 we noticed that Manila can return dates as ISO 8601 with an UTC offset that the sharedfilesystems module doesn't know how to parse.

We expect the `2006-01-02T15:04:05.999999` format [for JSONRFC3339MilliNoZ](https://github.com/gophercloud/gophercloud/blob/0721d75e876f20e0f3f9be83b3133fa3476ee992/results.go#L306), which is different from what Manila claims to return:

> The date and time stamp format is [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601):
> 
> CCYY-MM-DDThh:mm:ss±hh:mm
> 
> The ±hh:mm value, if included, returns the time zone as an offset from UTC.
> 
> For example, 2019-03-27T09:49:58-05:00.

Playing a little bit with it, it seems to break as soon as the date include an UTC offset: https://go.dev/play/p/pJh7jLgonx6. It looks like we're just being lucky.

**Comment by pierreprinetti** [comment]
@mandre is there outstanding work required besides #2496?

**Comment by mandre** [comment]
Yes, we identified the issue while reviewing https://github.com/gophercloud/gophercloud/pull/2496. We'll need to change the parsing to account for UTC offset, as this is something Manila can return, according to the docs.
This is also potentially an issue with other modules.
